lib/request-parser.js

```diff
@@ -7,10 +7,75 @@ const bodyParser = require("body-parser");
 const cookieParser = require("cookie-parser");
 const multer = require("multer");
 const tmp = require("tmp");
+const path = require("path");
+const util = require("./helpers/util");
 
 // Clean-up the temp directory, even if the app crashes
 tmp.setGracefulCleanup();
 
+/**
+ * Generates a middleware that parses multipart/form-data
+ */
+function generateMultipartFormDataMiddleware (options) {
+  options.allowAll = options.allowAll === undefined ? true : options.allowAll;
+  const uploader = multer(options);
+  return function multipartFormData (req, res, next) {
+    // Compatibility with old multer
+    req.body = req.body || {};
+    req.files = req.files || {};
+
+    // Get all the "file" params
+    if (util.isSwaggerRequest(req) && req.swagger.params.length > 0) {
+      const fileFields = [];
+      req.swagger.params.forEach((param) => {
+        if (param.in === "formData" && param.type === "file") {
+          fileFields.push({
+            name: param.name,
+            maxCount: 1
+          });
+        }
+      });
+
+      uploader.fields(fileFields)(req, res, (err) => {
+        if (err && err.code === "LIMIT_UNEXPECTED_FILE") {
+          next(); // let request-validator handle
```

Comment:
specifically for making 413 error work correctly in the request validation. I'm open to other approaches.

Reply:
This looks good.  Please call-out any behavioral changes in the changelog.